File: Android/fantastic_food/app/src/main/java/com/sander/fantasticfood/Model/UserProfile.java
package com.sander.fantasticfood.Model;

import androidx.annotation.Nullable;

import java.util.List;

public class UserProfile {
    private int id;
    private String email;
    private String passwordHash;
    private String weight;
    private String height;
    private List<Allergy> userAllergies;
    private List<MealPlan> userMealPlans;
    private List<Recipe> userFavourites;
    private List<Recipe> userRecipes;

    public UserProfile(int id, String email, String passwordHash, String weight, String height,
                       List<Allergy> userAllergies, List<MealPlan> userMealPlans,
                       List<Recipe> userFavourites, List<Recipe> userRecipes) {
        this.id = id;
        this.email = email;
        this.passwordHash = passwordHash;
        this.weight = weight;
        this.height = height;
        this.userAllergies = userAllergies;
        this.userMealPlans = userMealPlans;
        this.userFavourites = userFavourites;
        this.userRecipes = userRecipes;
    }

    public UserProfile(String email, String passwordHash) {
        this.email = email;
        this.passwordHash = passwordHash;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPasswordHash() {
        return passwordHash;
    }

    public void setPasswordHash(String passwordHash) {
        this.passwordHash = passwordHash;
    }

    public List<Allergy> getUserAllergies() {
        return userAllergies;
    }

    public void setUserAllergies(List<Allergy> userAllergies) {
        this.userAllergies = userAllergies;
    }

    public List<MealPlan> getUserMealPlans() {
        return userMealPlans;
    }

    public void setUserMealPlans(List<MealPlan> userMealPlans) {
        this.userMealPlans = userMealPlans;
    }

    public List<Recipe> getUserFavourites() {
        return userFavourites;
    }

    public void setUserFavourites(List<Recipe> userFavourites) {
        this.userFavourites = userFavourites;
    }

    public List<Recipe> getUserRecipes() {
        return userRecipes;
    }

    public void setUserRecipes(List<Recipe> userRecipes) {
        this.userRecipes = userRecipes;
    }

    public String getWeight() {
        return weight;
    }

    public void setWeight(String weight) {
        this.weight = weight;
    }

    public String getHeight() {
        return height;
    }

    public void setHeight(String height) {
        this.height = height;
    }

    @Override
    public boolean equals(@Nullable Object obj) {
        UserProfile other = (UserProfile) obj;
        if (this.getEmail() == other.getEmail()) {
            return true;
        } else {
            return false;
        }
    }
}

File: Flutter/fantastic_food/lib/domain/user_profile.dart
import 'package:fantastic_food/domain/allergy.dart';
import 'package:fantastic_food/domain/meal_plan.dart';
import 'package:fantastic_food/domain/recipe.dart';

class UserProfile {
  int id;
  String email;
  String passwordHash;
  String weight;
  String height;
  List<Allergy> userAllergies;
  List<MealPlan> userMealPlans;
  List<Recipe> userFavourites;
  List<Recipe> userRecipes;

  UserProfile(this.id, this.email, this.passwordHash, this.weight, this.height, 
    this.userAllergies, this.userMealPlans, this.userFavourites, this.userRecipes);

  UserProfile.byLogin({this.email, this.passwordHash});

  factory UserProfile.fromJson(dynamic json) {

    bool anyAllergies = false;
    bool anyMealPlans = false;
    bool anyFavourites = false;
    bool anyRecipes = false;

    if(json['userAllergies'] != null){
      anyAllergies = true;
    }
    if(json['userMealPlans'] != null) {
      anyMealPlans = true;
    }
    if(json['userFavourites'] != null){
      anyFavourites = true;
    }
    if(json['userRecipes'] != null) {
      anyRecipes = true;
    }

    return UserProfile(json['id'] as int, json['email'] as String, json['passwordHash'] as String, 
        json['weight'] as String, json['height'] as String, 
        anyAllergies? List<Allergy>.from(json['userAllergies'].map((model)=> Allergy.fromJson(model))): null,
        anyMealPlans? List<MealPlan>.from(json['userMealPlans'].map((model)=> MealPlan.fromJson(model))): null,
        anyFavourites? List<Recipe>.from(json['userFavourites'].map((model)=> Recipe.fromJson(model))): null,
        anyRecipes? List<Recipe>.from(json['userRecipes'].map((model)=> Recipe.fromJson(model))): null);
  }


  Map<String, dynamic> toJson() {
    return {
      "id": this.id,
      "email": this.email,
      "passwordHash": this.passwordHash,
      "weight": this.weight,
      "height": this.height,
      "userAllergies": this.userAllergies,
      "userMealPlans": this.userMealPlans,
      "userFavourites": this.userFavourites,
      "userRecipes": this.userRecipes
    };
  }
}